test/select_tests.cpp

```diff
@@ -805,3 +805,123 @@ TEST(IntervalLiteral) {
+
+  stmt = (SelectStatement*)result.getStatement(0);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::NoWait);
+
+  stmt = (SelectStatement*)result.getStatement(1);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::NoWait);
+
+  stmt = (SelectStatement*)result.getStatement(2);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForNoKeyUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::NoWait);
+
+  stmt = (SelectStatement*)result.getStatement(3);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForKeyShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::NoWait);
+
+  stmt = (SelectStatement*)result.getStatement(4);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::SkipLocked);
+
+  stmt = (SelectStatement*)result.getStatement(5);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::SkipLocked);
+
+  stmt = (SelectStatement*)result.getStatement(6);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForNoKeyUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::SkipLocked);
+
+  stmt = (SelectStatement*)result.getStatement(7);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForKeyShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::SkipLocked);
+}
+TEST(LockingClauseWithTableReference) {
+  SelectStatement* stmt;
+  TEST_PARSE_SQL_QUERY(
+      "SELECT * FROM t WHERE a = 10 FOR UPDATE OF t;"
+      "SELECT * FROM t, c WHERE t.a = 10 FOR SHARE OF t,c;"
+      "SELECT * FROM t, c WHERE t.a = 10 FOR NO KEY UPDATE OF t,c NOWAIT;",
+      result, 3);
+
+  stmt = (SelectStatement*)result.getStatement(0);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::None);
+  ASSERT_STREQ(stmt->lockings->at(0)->tables->at(0), "t");
+
+  stmt = (SelectStatement*)result.getStatement(1);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForShare);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::None);
+  ASSERT_STREQ(stmt->lockings->at(0)->tables->at(0), "t");
+  ASSERT_STREQ(stmt->lockings->at(0)->tables->at(1), "c");
+
+  stmt = (SelectStatement*)result.getStatement(2);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockMode, RowLockMode::ForNoKeyUpdate);
+  ASSERT_EQ(stmt->lockings->at(0)->rowLockWaitPolicy, RowLockWaitPolicy::NoWait);
+  ASSERT_STREQ(stmt->lockings->at(0)->tables->at(0), "t");
+  ASSERT_STREQ(stmt->lockings->at(0)->tables->at(1), "c");
+}
+
+TEST(MultipleLockingClause) {
```

Comment:
Please also check the number of elements in `lockings` and add a statement with multiple locking clauses where one clause has no tables specified, and two other clauses have tables specified but the tables and numbers of specified tables are different.